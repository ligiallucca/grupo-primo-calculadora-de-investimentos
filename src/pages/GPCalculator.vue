<script setup lang="ts">
import { watch } from 'vue';
import { store } from '@/store';
import GPForm from '../components/GPForm.vue';
import GPResults from '../components/GPResults.vue';
import type { InvestmentFormData } from '@/interfaces/InvestmentFormData';

watch(() => store.investmentFormData, () => getResults() )

const formData = store.investmentFormData;
const taxes = store.taxes

const getResults = () => {
  console.log("ðŸš€ ~ getResultssssssssssssssssssssssssssssssssss ~ getResults:", getResults)
const totalArca = calcTotalAmount(formData, taxes.arcaTax); 
  console.log("ðŸš€ ~ onSubmit ~ totalArca:", totalArca);
  const totalSelic = calcTotalAmount(formData, taxes.selicTax);
  console.log("ðŸš€ ~ onSubmit ~ totalSelic:", totalSelic);
  store.results.arcaResult = totalArca;
  store.results.selicResult = totalSelic;
}


const calcTotalAmount = (formData: InvestmentFormData,  tax: number): number =>  {
  const simpleInterestAmount = calcSimpleInterest(formData, tax);
  const compoundInterestAmount = calcCompoundInterest(formData, tax);
  const totalAmonut = simpleInterestAmount + compoundInterestAmount;
  console.log("ðŸš€ ~ calcTotalAmount ~ totalAmonut:", totalAmonut);
  return totalAmonut;  
}

const calcSimpleInterest = (formData: InvestmentFormData, tax: number, ) : number => {
  console.log('simple', formData, tax)
  const monthlyTax = tax / 12;
  console.log("ðŸš€ ~ calcSimpleInterest ~ monthlyTax:", monthlyTax);
  const numberOfMonthsToNumber = parseFloat(formData.numberOfMonths);
  console.log("ðŸš€ ~ calcSimpleInterest ~ numberOfMonthsToNumber:", numberOfMonthsToNumber)
  const initialInvestimentToNumber = parseFloat(formData.initialInvestment);
  console.log("ðŸš€ ~ calcSimpleInterest ~ initialInvestimentToNumber:", initialInvestimentToNumber)
  
  //qto rende so o valor inical em qtdMeses
  const simpleInterestResult = initialInvestimentToNumber * (1 + monthlyTax) ** numberOfMonthsToNumber; 

  return simpleInterestResult;
}

const calcCompoundInterest = (formData:InvestmentFormData, tax: number ) : number => {
  const monthlyTax = tax / 12;
  const numberOfMonthsToNumber = parseFloat(formData.numberOfMonths);
  const monthlyInvestmentToNumber = parseFloat(formData.monthlyValue);
  
  //qto rende so o valor inical em qtdMeses
  const compoundInterestResult = monthlyInvestmentToNumber * ((1 + monthlyTax) ** numberOfMonthsToNumber -1) / monthlyTax; 
  return compoundInterestResult;
}
</script>

<template>
  <main>
    <GPForm @formSubmited="getResults()"></GPForm>
    <GPResults></GPResults>
  </main>
</template>

<style lang="scss" scoped>
main {
  padding: 96px 120px;
  display: flex;
  margin: 0 auto;
  width: 100%;
  gap: 140px;
}
</style>