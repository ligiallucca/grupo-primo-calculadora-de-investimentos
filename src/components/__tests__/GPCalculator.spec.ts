import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import GPCalculator from '@/pages/GPCalculator.vue'


describe('GPHeader', () => {

    it('renders properly', () => {
        const wrapper = mount(GPCalculator)
        expect(wrapper.html).toMatchSnapshot()
    })
})