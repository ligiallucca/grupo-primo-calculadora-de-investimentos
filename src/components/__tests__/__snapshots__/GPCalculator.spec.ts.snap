// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GPHeader > renders properly 1`] = `[Function]`;
